lib/admin.rb

```diff
@@ -0,0 +1,90 @@
+require 'date'
+require_relative 'reservation'
+require_relative 'block_room'
+
+class Admin
+  attr_reader :roomlist, :reservations, :price, :blocks
+
+  def initialize
+    @roomlist = []
+    @reservations = []
+    @blocks = []
+
+    20.times do |i|
+      @roomlist << (i + 1)
+    end
+  end
+
+  def add_reservation(check_in, check_out, room_id)
+    available_rooms = available_rooms(check_in, check_out)
+
+    if !(available_rooms.include?(room_id))
+      raise ArgumentError.new("#{room_id} is booked for these dates")
+    end
+
+    if available_rooms.count == 0
```

Comment:
Ummm.. if there were no rooms then `available_rooms` would not include `room_id`.